kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -142,21 +146,22 @@ protected void doInit() {
             System.exit(1);
           });
       this.shutdownDuration = getShutdownDuration();
+      this.initialDelay = InitialDelayConfigProvider.getInstance().getInitialDelay(streamsConfig);
+      this.isSleeping = true;
```

Comment:
By default, `isSleeping` should not be true (to handle cases when config is not set). It should be set to true inside `handleInitialDelay()` just before we start sleeping, and reset to false when sleep ends.

Reply:
Done. I had it this way. Changed it back to align with your suggestion